tests/queries/tests.py

```diff
@@ -82,6 +83,30 @@ def setUpTestData(cls):
         Cover.objects.create(title="first", item=i4)
         Cover.objects.create(title="second", item=cls.i2)
 
+    def test_explain(self):
+        querysets = [
+            Tag.objects.filter(name='test').all(),
+            Tag.objects.filter(name='test').select_related("parent"),
+            Tag.objects.filter(name='test').prefetch_related("children"),
+            Tag.objects.filter(name='test').annotate(Count('children')),
+            Tag.objects.filter(name='test').values_list('name'),
+            Tag.objects.order_by().union(Tag.objects.order_by().filter(name='test')),
+            Tag.objects.all().select_for_update().filter(name='test')
+        ]
+        supported_formats = connection.features.supported_explain_formats
+        all_formats = (None,) + tuple(supported_formats) + tuple(f.lower() for f in supported_formats)
+        for idx, queryset in enumerate(querysets):
+            for verbose, output_format in itertools.product((True, False), all_formats):
+                with self.subTest(verbose=verbose, format=output_format, queryset=idx):
+                    result = queryset.explain(format=output_format, verbose=verbose)
+                    self.assertIsInstance(result, str)
+                    self.assertTrue(result)
+
+    def test_explain_unknown_format(self):
+        qs = Tag.objects.filter(name='test').all()
+        with self.assertRaises(ValueError):
```

Comment:
Please test the value of the message. Since it's variable, depending on the backend, I think it's fine to use `supported_explain_formats` with a check if it has a value.

Reply:
Can you elaborate? I'm not sure what you mean, how would I use `supported_explain_formats` in this context?